### PR TITLE
[serve] Flip default for `RAY_SERVE_ENABLE_NEW_HANDLE_API` to `1`

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Changes the default for the handle API to use `DeploymentHandle`. This will be a breaking API change. A warning was logged to inform users since Ray 2.7 (in addition to a warning on the documentation) and this will land in Ray 2.9.

Users can continue to use the existing handle API by setting the environment variable `RAY_SERVE_ENABLE_NEW_HANDLE_API=0` or using `handle.options(use_new_handle_api=False)`.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/handle.py [hunk 1]
```diff
@@ -228,12 +228,11 @@ def _remote(
...
             warnings.warn(
-                "Ray 2.7 introduces a new `DeploymentHandle` API that will "
-                "replace the existing `RayServeHandle` and `RayServeSyncHandle` "
-                "APIs in a future release. You are encouraged to migrate to the "
-                "new API to avoid breakages in the future. To opt in, either use "
-                "`handle.options(use_new_handle_api=True)` or set the global "
-                "environment variable `export RAY_SERVE_ENABLE_NEW_HANDLE_API=1`. "
+                "`DeploymentHandle` is now the default handle API. You can continue "
+                "using the existing `RayServeHandle` and `RayServeSyncHandle` APIs "
+                "by calling `handle.options(use_new_handle_api=True)` or setting the "
```
Comment (shrekris-anyscale) [inline python/ray/serve/handle.py:233]:
```suggestion
                "by calling `handle.options(use_new_handle_api=False)` or setting the "
```

File: python/ray/serve/handle.py [hunk 2]
```diff
@@ -228,12 +228,11 @@ def _remote(
...
+                "global environment variable `RAY_SERVE_ENABLE_NEW_HANDLE_API=1`, "
```
Comment (shrekris-anyscale) [inline python/ray/serve/handle.py:234]:
```suggestion
                "global environment variable `RAY_SERVE_ENABLE_NEW_HANDLE_API=0`, "
```

**Comment by edoakes** [comment]
@ray-project/ray-serve this is ready for review now (tests are passing).

File: python/ray/serve/tests/common/remote_uris.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# flake8: noqa
```
[the remark this answers is not in the document]
Comment (GeneDer) [reply python/ray/serve/tests/common/remote_uris.py:1]: Nice!

**Review by GeneDer — APPROVED** [review]
Great work! thanks Ed! Just some non-blocking minor comments, changes LGTM!

File: python/ray/serve/_private/client.py [hunk 4]
```diff
@@ -510,16 +510,22 @@ def get_handle(
             handle = DeploymentHandle(
                 deployment_name,
                 app_name,
+                # Only used when users convert this back to deprecated handle types.
+                sync=sync,
```
[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/serve/_private/client.py:514]: This is still needed if users call `.options(use_new_handle_api=False)`

File: python/ray/serve/drivers.py [hunk 5]
```diff
@@ -64,7 +64,6 @@ async def handle_request(inp=Depends(http_adapter)):
                 endpoint_create_func()
 
         else:
-            assert isinstance(dags, (RayServeDAGHandle, RayServeHandle))
             if isinstance(dags, RayServeHandle):
                 dags = dags.options(use_new_handle_api=True)
```
[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/serve/drivers.py:68]: This is needed if users set the global env variable feature flag off

File: python/ray/serve/_private/deployment_function_executor_node.py [hunk 6]
```diff
@@ -57,8 +57,10 @@ def _execute_impl(self, *args, **kwargs) -> ObjectRef:
         receive whatever this method returns. We return a handle here so method
         node can directly call upon.
         """
-        return self._deployment_function_handle.remote(
-            *self._bound_args, **self._bound_kwargs
+        return (
+            self._deployment_function_handle.options(use_new_handle_api=True)
```
[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/serve/_private/deployment_function_executor_node.py:61]: This is needed if users set the global env variable feature flag off

File: python/ray/serve/tests/common/remote_uris.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shrekris-anyscale) [reply python/ray/serve/tests/common/remote_uris.py:1]: Thanks!

File: python/ray/serve/handle.py [hunk 7]
```diff
@@ -513,8 +513,9 @@ async def _to_object_ref_or_gen(
     def _to_object_ref_or_gen_sync(
         self,
         _record_telemetry: bool = True,
+        _allow_running_in_asyncio_loop: bool = False,
```
Comment (shrekris-anyscale) [inline python/ray/serve/handle.py:516]: Why do we need to add this parameter? When would it be set to `True`/`False`?

**Review by shrekris-anyscale — APPROVED** [review]

File: python/ray/serve/handle.py (hunk 7 above)
Comment (edoakes) [reply python/ray/serve/handle.py:516]: disables the exception for internal usage

**Comment by can-anyscale** [comment]
w00t, @edoakes , I think this might break the serve dashboard tests on postmerge https://buildkite.com/ray-project/postmerge/builds/1570.

The serve dashboard test was not run during premerge which we should fix as well.

**Pull request merged**